src/components/notifications/SMTPConfigModal.tsx

```diff
@@ -89,41 +89,50 @@ export const SMTPConfigModal = ({
         }
     }
 
-    const getAllFieldsValidated = () => {
+    const validateSave = () => {
         const { configName, host, port, authUser, authPassword, fromEmail } = form
+        setFormValid((prevValid) => ({
+            ...prevValid,
+            configName: validateKeyValueConfig(ConfigurationFieldKeys.CONFIG_NAME, form.configName),
+            host: validateKeyValueConfig(ConfigurationFieldKeys.HOST, form.host),
+            port: validateKeyValueConfig(ConfigurationFieldKeys.PORT, form.port),
+            authUser: validateKeyValueConfig(ConfigurationFieldKeys.AUTH_USER, form.authUser),
+            authPassword: validateKeyValueConfig(ConfigurationFieldKeys.AUTH_PASSWORD, form.authPassword),
+            fromEmail: validateKeyValueConfig(ConfigurationFieldKeys.FROM_EMAIL, form.fromEmail),
+        }))
+        setForm((prevForm) => ({ ...prevForm, isLoading: false }))
         return (
-            !!configName &&
-            !!host &&
-            !!port &&
-            !!authUser &&
-            !!authPassword &&
-            !!fromEmail &&
-            getFormValidated(isFormValid, fromEmail)
+            validateKeyValueConfig(ConfigurationFieldKeys.CONFIG_NAME, configName).isValid &&
```

Comment:
Why are again validationg the same. we should use the value from line no 94 setter